activerecord/lib/active_record/connection_adapters/mysql/column.rb

```diff
@@ -5,7 +5,7 @@ class Column < ConnectionAdapters::Column # :nodoc:
         delegate :extra, to: :sql_type_metadata, allow_nil: true
 
         def unsigned?
-          /\bunsigned\z/.match?(sql_type)
+          /\bunsigned\b/.match?(sql_type)
```

Comment:
@kamipo The enum unsigned test will fail due to this, I think we need special treatment for enum.

Something like this:

```ruby
        def unsigned?
          if (/^enum/).match?(sql_type)
            actual_sql_type = sql_type[/^enum\((.+)\)(.+)/i, 2]
          else
            actual_sql_type = sql_type
          end

          /\bunsigned\b/.match?(actual_sql_type)
        end
```